interfaces/builtin/home.go

```diff
@@ -63,14 +76,56 @@ owner /run/user/[0-9]*/gvfs/{,**} r,
 owner /run/user/[0-9]*/gvfs/*/**  w,
 `
 
+const homeConnectedPlugAppArmorWithAllRead = `
+# Allow non-owner read to non-hidden and non-snap files and directories
+@{HOME}/               r,
+@{HOME}/[^s.]**        r,
+@{HOME}/s[^n]**        r,
+@{HOME}/sn[^a]**       r,
+@{HOME}/sna[^p]**      r,
+@{HOME}/snap[^/]**     r,
+@{HOME}/{s,sn,sna}{,/} r,
+`
+
+type homeInterface struct {
+	commonInterface
+}
+
+func (iface *homeInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	// It's fine if 'read' isn't specified, but if it is, it needs to be
+	// either 'all' or 'owner'
+	if r, ok := plug.Attrs["read"]; ok {
+		_, ok = r.(string)
+		if !ok || (r != "all" && r != "owner") {
+			return fmt.Errorf(`home plug requires the "read" be either 'all' or 'owner'`)
+		}
+	}
+
+	return nil
+}
+
+func (iface *homeInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	var read string
+	_ = plug.Attr("read", &read)
+	// 'owner' is the standard policy
+	spec.AddSnippet(homeConnectedPlugAppArmor)
+
+	// 'all' grants standard policy plus read access to home without owner
+	// match
+	if read == "all" {
+		spec.AddSnippet(homeConnectedPlugAppArmorWithAllRead)
+	}
+	return nil
+}
+
 func init() {
-	registerIface(&commonInterface{
+	registerIface(&homeInterface{commonInterface{
```

Comment:
I must say I'm really happy we're doing this. It makes to-and-from common vs custom interface transitions so much easier than before.